server.go

```diff
@@ -3728,6 +3736,49 @@ func (s *server) connectToPeer(addr *lnwire.NetAddress,
 	s.OutboundPeerConnected(nil, conn)
 }
 
+// updateConnRequests checks to see if there are any existing connection
+// requests active for the given pubKey and if there are then it checks that
+// there are connection requests active for all the given addresses. If any are
+// missing then new connection requests are created.
+func (s *server) updateConnRequests(pubKey *btcec.PublicKey, addrs []net.Addr) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Check if there are currently any connection requests for this node.
+	pubStr := string(pubKey.SerializeCompressed())
+	reqs, ok := s.persistentConnReqs[pubStr]
+	if !ok {
+		return
+	}
+
+	connSet := make(map[string]bool)
+	for _, req := range reqs {
+		connSet[req.Addr.String()] = true
+	}
+
+	for _, addr := range addrs {
+		lnAddr := &lnwire.NetAddress{
+			IdentityKey: pubKey,
+			Address:     addr,
+		}
+
+		if connSet[lnAddr.String()] {
+			continue
+		}
+
+		connReq := &connmgr.ConnReq{
+			Addr:      lnAddr,
+			Permanent: true,
+		}
+
+		s.persistentConnReqs[pubStr] = append(
```

Comment:
What about `s.persistentPeersBackoff` and `s.persistentRetryCancels`, do we need to update them too?

Reply:
latest change does update `s.persistentPeersBackoff` but i dont htink that  `s.persistentRetryCancels` needs to be updated